dm/templates/route/single_route.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2019 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This template creates a custom route."""
+
+
+from hashlib import sha1
+import json
+
+
+def generate_config(context):
+    """ Entry point for the deployment resources. """
+
+    properties = context.properties
+    project_id = properties.get('project', context.env['project'])
+
+    # Set the common route properties.
+    res_properties = {
+        'network': properties['network'],
+        'project': project_id,
+        'tags': properties['tags'],
+        'priority': properties['priority'],
+        'destRange': properties['destRange']
+    }
+
+    # Check the route type and fill out the following fields:
+    if properties.get('routeType') == 'instance':
+        instance_name = properties.get('instanceName')
+        zone = properties.get('zone', '')
+        res_properties['nextHopInstance'] = generate_instance_url(
+            project_id,
+            zone,
+            instance_name
+        )
+    elif properties.get('routeType') == 'gateway':
+        gateway_name = properties.get('gatewayName')
+        res_properties['nextHopGateway'] = generate_gateway_url(
+            project_id,
+            gateway_name
+        )
+    elif properties.get('routeType') == 'vpntunnel':
+        vpn_tunnel_name = properties.get('vpnTunnelName')
+        region = properties.get('region', '')
+        res_properties['nextHopVpnTunnel'] = generate_vpn_tunnel_url(
+            project_id,
+            region,
+            vpn_tunnel_name
+        )
+
+    optional_properties = [
+        'nextHopIp',
+        'nextHopInstance',
+        'nextHopNetwork',
+        'nextHopGateway',
+        'nextHopVpnTunnel',
+    ]
+
+    for prop in optional_properties:
+        if prop in properties:
+            res_properties[prop] = properties[prop]
+
+    name = properties['name']
+    resources = [
+        {
+            'name': name,
```

Comment:
Please use context.env['name'] and add name to properties

Reply:
You have two names here:
- DM resource name
- real resource name

Previously we were using real resource names as DM resource names, that is incorrect. DM resource name should always be context['name'], while real resource name can be overridden by user

Reply:
I don't see properties->name being set